torch/_dynamo/side_effects.py

```diff
@@ -36,9 +36,8 @@ class MutableSideEffects(MutableLocalBase):
     the graph runs.
     """
 
-    def __init__(self, source: Source, is_modified: bool = False):
+    def __init__(self, is_modified: bool = False):
```

Comment:
Not relevant to this line and can be done in another PR. Can we move `MutableLocal` and `MutableLocalBase` in this file? Good to consolidate them here, and then add a comment what different types of subclasses mean with some examples.

Reply:
Hmm, moving `MutableLocal` could be messy given so many other files import it.

Reply:
Yeah I also thought it's a tad odd to have these mutability marker subclasses in different files, as it hinders reasoning about the cases. I'll see what I can do.